---
title: Bioinformatics Overview
last_modified_at: 2018-08-31
---

Bioinformatics refers to the computational analysis of raw biological
data, typically those generated within the context of a scientific experiment or
research project. One of the most rapidly expanding and heavily
used application of bioinformatics is genomic sequence analysis,
in which some form of high-throughput, large scale data collection instrument
(such as a genome sequencer or a DNA microarray) is used to collect
a large amount of information about such biological entities as
genes or transcripts, etc.  Large scale data generation approaches like this can be referred to as molecular profiling, and can be applied to a variety of biological and biomedical research contexts from model organism studies (e.g., yeast strains) to human specimens in the context of a clinical assay.  

## [Best Practices for Research Data Management]({{ site.baseurl }}/bioinformatics/dm_overview/)
While the primary focus of research is the processes more clearly involved in the scientific endeavor, often the importance of knowing some degree of best practices for various data oriented issues is understated. Here we aim to summarize the essential points of the best practices for the following topics here at the Fred Hutch.

- [Data Storage]({{ site.baseurl }}/bioinformatics/dm_storage/)
- [Data Ingestion and Public Datasets]({{ site.baseurl }}/bioinformatics/dm_ingest/)
- [Data Archiving]({{ site.baseurl }}/bioinformatics/dm_archive/)

## [Programming]({{ site.baseurl }}/bioinformatics/prog_overview/)
While there are many types of programming languages various software used in scientific research are written in, there are a handful of specific languages that are commonly used in the process of doing a wide range of research tasks. We will introduce the most common here.

- [R, RStudio]({{ site.baseurl }}/bioinformatics/prog_R/)
- [Python]({{ site.baseurl }}/bioinformatics/prog_python/)
- [Unix/Bash and Other Common Languages]({{ site.baseurl }}/bioinformatics/prog_unixbash/)

## [Managing Code and Computing]({{ site.baseurl }}/bioinformatics/compute_overview/)
Researchers are more and more likely to need to analyze raw data sets using some sort of analysis process before they can be interpreted in the context of the scientific question. Doing this work requires knowledge of what compute resources are available, how one might interact with the code they write, attention to the need for version control and workflow creation.  
- [Fred Hutch Compute Resources]({{ site.baseurl }}/bioinformatics/compute_resoverview/)
- [GitHub, Versioning and Sharing]({{ site.baseurl }}/bioinformatics/compute_github/)
- [Notebooks and UIs]({{ site.baseurl }}/bioinformatics/compute_uis/)
- [Docker]({{ site.baseurl }}/bioinformatics/compute_docker/)


## [Training and References]({{ site.baseurl }}/bioinformatics/howto_training/)
These sections include a resource list of good opportunities for training from on-campus options to web-based learning options.  Additionally, the Research Topic Articles are a work in progress, where we can host short demo's or articles about common questions or approaches as written by Fred Hutch based researchers and staff.  

- [Training Opportunities](({{ site.baseurl }}/bioinformatics/howto_training/))
- [Research Topic Articles]({{ site.baseurl }}/bioinformaticsdemos/)

