---
title: Genomics Platforms and Data Types
---

This guide highlights some of the genomics platforms available through the Genomics Core at Fred Hutch.  This guide is intended to give general context to each platform.

## Sequencing Based Platforms
Sequencing based platforms currently available via the Genomics Core at the Fred Hutch include the Illumina sequencers (HiSeq/MiSeq) and the Pacific Biosciences Long Range sequencer.  These platforms can be used to sequence a variety of assay material types via different library preparation processes. In the RNA or DNA Approaches pages, we discuss different options for the creation of libraries for sequencing from either nucleic acid type and for different research questions.  Choosing the appropriate assay material QC and library preparation reagents depends in part on how the libraries will be sequenced.  Thus is it important to verify that all phases of the process are using compatible techniques.  

### Illumina Sequencers
The Illumina sequencers are high read number, short read sequencers that provide a range of sequencing capabilities for many different upstream library types. The primary approach of these sequencers is to sequence at high numbers, individual fragments of DNA generated by the library preparation process, then to reconstruct the sequences in the mixture by, for example, aligning the sequences of the short reads to a reference, or other bioinformatic approach. The Illumina HiSeq at the Fred Hutch currently has two options:  the High Output mode or the Rapid Run mode.  The basic details for each mode along with the corresponding information for the MiSeq are included below.  We have provided cost ranges to provide a basic idea of the relative costs of these sequencing platforms, but the exact costs of a sequencing run will depend on the read lengths, whether the sequencing is paired end or single end, as well as your affiliation (Fred Hutch vs external).

---
Sequencer | Mode | Read Lengths | Reads per Lane | Lanes per Run |  Cost per Lane ($)
--- | --- | --- | --- | --- | ----
HiSeq | High Output | 50, 100, 125 | 250M | 8\* | $900-$4,600
HiSeq | Rapid Run | 50, 100, 150, 200, 250 | 150M | 2 | $900-$4,600
MiSeq | Version 2 Reagents | up to 250 | 15M | 1 | $1,000-$2,100
MiSeq | Version 3 Reagents | up to 300 | 25M | 1 | $1,200-$2,700

\* High Output mode requires all 8 lanes to be run simultaneously, no partial runs.

---
When deciding how much sequencing is needed for a set of libraries to provide sufficient read depth (number of reads per genomic location in the genome covered in the library), issues such as the intended data type, sample type and quality, library preparation type, number of total samples, and the applicability of multiplexing approaches need to be considered.  Consulting with the Genomics Core can help provide more clarity for individual projects.  

[Illumina Sequencing](https://www.illumina.com/techniques/sequencing.html){:target="_blank"}<!--_-->

### Pacific Biosciences (PacBio) Long Range Sequencer
The PacBio SMRT sequencer works differently than the Illumina sequencers in that the read length is not specified by the platform, but is limited by the library itself, with an associated reduction in confidence of the sequence as reads get longer and longer.  However, instead of being limited to sequencing only fragments of DNA, PacBio sequencing can provide long stretches of sequencing data that occur in the same fragment.  This allow for analyses such as full length isoform discovery, de novo small genome sequencing, assessing structural variants/translocations, and allele phasing.  On average the PacBio sequencer aims to provide up to 15kb of read length.  


---
#### Overview
- Individual SMRT cells (individual sequencing unit) cost is in the range of $900 to $2,400
- 300,000 reads per SMRT cell
- Insert size for sequencing of 200bp minimum up to 40kb fragments
- Library prep
  - Reagents purchased by lab from PacBio and prepped by lab, brought to Genomics as completed library (see PacBio website for more info)
  - OR
  - Genomics Lab library prep and service fee to process a QC'd sample into a Pac Bio library:
    - For amplicon library prep up to 5kb:  $300-500 per sample
    - For large insert library prep up to 20kb:  $400-700 per sample
- Multiple multiplexing schemas (in-line or ligated) - discuss details with Genomics Lab to plan the approach.  
---

[PacBio SMRT Sequencing](http://www.pacb.com/smrt-science/smrt-sequencing/){:target="_blank"}<!--_-->


## Array Based Platforms
 Microarrays are a sometimes less costly option that can in some cases be substituted for a wide variety of sequencing types; for example, there are SNP, gene expression, and whole exome arrays. While microarrays are not useful for discovery of novel targets, for well-established targets, assay chemistries and data analysis pipelines are well-vetted.  A discussion with the Genomics Core can be useful in helping you decide the best technologies for your work.

### Single Nucleotide Polymorphism (SNP) Arrays or Methylation Arrays
The Genomics lab is equipped to run all Illumina genotyping and methylation beadchips.  Beadchip kits are sold in a variety of sample size kits and it is important to plan total sample sizes as well as randomizing batches to minimize batch-specific bias. Beadchip kits need to be purchased by the investigator and should be drop-shipped to the Genomics lab.  The Genomics lab then charges a fee for processing which covers all non-Illumina supplies, reagents, and labor.  Investigators should plan to provide genomic DNA for genotyping arrays, which has been quantified by a dsDNA specific method such as picogreen.  For methylation arrays, investigators may submit either genomic DNA or bisulfite converted DNA.  Again, the starting DNA should be quantified by picogreen.  

#### Costs:
Genotyping BeadChip kits vary in price ($50/sample - $600/sample), depending on content.  Our processing fee is $100/sample, regardless of the beadchip kit.

The EPIC Methylation beadchip kit is ~$375/sample.  Our processing fee for DNA that is already bisulfite converted is $100/sample.  If conversion is needed, we offer that service for an additional $50/sample.

[Illumina Microarrays](https://www.illumina.com/techniques/microarrays.html){:target="_blank"}<!--_-->

### Nanostring Hybridization Arrays for Gene Expression
Typical use case, advantages and disadvantages

[Nanostring](https://www.nanostring.com){:target="_blank"}<!--_-->


## Library Preparation Reagents and Methods
The choice of genomics platform will dictate the needs of the library preparation method.  It is important to understand how library preparation can impact the final data, for example if biases towards detection of specific types of nucleic acids are introduced.  Meeting with an NGS specialist at the Genomics Core will help to guide your path.  

### Library Preparation for Sequencing
The four main steps in preparing RNA or DNA for NGS analysis are:

1. fragmenting and/or sizing the target sequences to a desired length (via physical, enzymatic, and chemical methods)
2. converting target to double-stranded DNA (if RNA)
3. attaching oligonucleotide adapters to the ends of target fragments--these adaptors are multi-purpose. They are used to index/barcode the fragments, and allow the fragments to be attached to the flow cell of the sequencer (where the sequencing of the fragments occurs)
4. quantitating the final library product for sequencing

### 10x Genomics Single Cell Library Preparation System
To obtain single cell gene expression data from RNA-seq, the Genomics lab uses the 10x Genomics Single Cell Expression platform.  Starting with a cell suspension, this process partitions cells into droplets for cDNA library preparation.  After library prep, the droplets are pooled, then sequenced on an Illumina sequencer. Unique molecular identifiers used in the library prep allow the sequencing results to be computationally traced back to individual cells.
[10X Genomics](https://www.10xgenomics.com){:target="_blank"}<!--_-->

### Overview of the Sequencing Process
1. the adapter-ligated DNA library is loaded onto a flowcell.
2. The fragments are hybridized to the flow cell surface.
3. Each bound fragment is amplified into a cluster. This step is known as bridge amplification.
4. Fluorescently-labeled nucleotides and sequencing reagents are added; the flow cell is fluorescently imaged after the incorporation of each nucleotide. The color of the fluorescent dyes identifies which base was incorporated. This cycle of nucleotide incorporation and imaging is repeated n times for a n-reads of sequence.


## Data types and sizes for common platforms
What file-types are generated from each of the platforms listed above, and what is the expected file size per sample
### Illumina
More to come.

### PacBio
More to come. For each SMRTcell we provide raw PacBio subreads in BAM format. We also provide consensus reads, in Fastq or Fasta format, produced with the CCS2 workflow. More specialized applications, such as small genome assembly with HGAP4 or long amplicon analysis (LAA2), may be available on a fee-for-service basis.

### Illumina Microarrays
More to come.

### Nanostring
More to come.

### 10x Genomics
More to come.

### Available Resources
 - [The Hutch Genomics Core:](https://sharedresources.fredhutch.org/core-facilities/genomics){:target="_blank"}<!--_-->
The genomics core is VERY helpful if you need guidance about reagent and platform choice for your samples.
- Genohub has a compendium of library prep kits, organized by NGS application type [here](https://genohub.com/ngs-library-preparation-kit-guide/){:target="_blank"}<!--_-->.
- Illumina has a few interactive methods guides to help you find the most appropriate [library prep reagents](https://www.illumina.com/techniques/sequencing/ngs-library-prep.html){:target="_blank"}<!--_--> and [sequencing methods](https://www.illumina.com/science/sequencing-method-explorer.html){:target="_blank"}<!--_--> to use in your experiments.
- A helpful reference in considering library prep methods is [Ordoukhanian's 2014 paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4351865/){:target="_blank"}<!--_-->.
